packages/mobile/src/send/utils.ts

```diff
@@ -0,0 +1,269 @@
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import { FeeType } from 'src/fees/actions'
+import { E164NumberToAddressType, RecipientVerificationStatus } from 'src/identity/reducer'
+import { Recipient, RecipientKind } from 'src/recipients/recipient'
+import { SecureSendPhoneNumberMapping } from 'src/send/reducers'
+import { TransactionDataInput } from 'src/send/SendAmount'
+import { ConfirmationInput } from 'src/send/SendConfirmation'
+import Logger from 'src/utils/Logger'
+
+export const getAddressFromPhoneNumber = (
+  e164Number: string,
+  e164NumberToAddress: E164NumberToAddressType,
+  secureSendPhoneNumberMapping: SecureSendPhoneNumberMapping
+): string | null | undefined => {
+  const TAG = 'send/utils'
+  try {
+    const addresses = e164NumberToAddress[e164Number]
+
+    // If address is null (unverified) or undefined (in the process
+    // of being updated) then just return that falsy value
+    if (!addresses) {
+      return addresses
+    }
+
+    // If there are multiple addresses, need to determine which to use
+    if (addresses.length > 1) {
+      // Check if the user has gone through Secure Send and validated a
+      // recipient address
+      const validatedAddress = secureSendPhoneNumberMapping[e164Number]
+        ? secureSendPhoneNumberMapping[e164Number].address
+        : undefined
+
+      // If they have not, they shouldn't have been able to
+      // get to this point
+      if (!validatedAddress) {
+        throw new Error(
+          'Multiple addresses but none were validated. Should have routed through Secure Send.'
+        )
+      }
+
+      return validatedAddress
+    }
+
+    // Normal case when there is only one address in the mapping
+    return addresses[0]
+  } catch (error) {
+    Logger.error(TAG, error.message)
```

Comment:
What should happen on error? Should we propagate it? Return null?

Reply:
That's a good question and I am not sure what the answer is. This is a true error in the sense that it really should never happen. If it does, that means we are not using the function correctly (i.e., it's being called before the SecureSend flow has collected the data).

Both `null` (unverified) and `undefined` (loading) have a reserved meaning here. How do we feel about making this a breaking error? If so, what's the best way to implement that?

Reply:
Yeah if it's a true error and we don't have a clean way to recover, just let it propagate and our React error boundary will end up catching it